research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+</p>
+
+<h3 id="showing-popup-steps">
+  Showing a <code>popup</code> element steps
+</h3>
+
+1. Run the [hiding currently-shown `popup` elements steps](#hiding-current-popups).
+2. Add the _subject_ to _subject_'s
+   [node document](https://dom.spec.whatwg.org/#concept-node-document)'s
+   [top layer](https://fullscreen.spec.whatwg.org/#top-layer), at the top of the stack.
+3. Set the `open` IDL attribute on the _subject_ to `true`.
+
+<h4 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h4>
+
+Let _candidate subject_ be the new candidate `popup` element to be shown.
+If applicable, let _activating element_ be the element whose `popup` attribute specifies the ID of
+the _candidate subject_, and which has been activated by the user.
+If applicable, let the _anchor element_ be the element whose ID is referenced by the `anchor`
+specified on the _candidate subject_.
+
+Starting at the bottom of the top-layer stack: for each `popup` element which has an `open` IDL
+attribute set to `true`, and which is not the _candidate subject_:
+
+1. Let _subject_ be the current `popup` element in the top-layer stack.
+2. Walk the DOM to determine if the _subject_ is an ancestor to the _candidate subject_, the
+   _activating element_, or the _anchor element_.
+3. If the _subject_ is an ancestor, then return.
+4. Otherwise, for this _subject_ and any other `popup`s higher in the stack, run the
+   [hiding a `popup` element steps](#hiding-current-popups).
+
+<p class="todo">
+  Step #2 probably needs to be more precise, and more specific about Shadow/light DOM
+</p>
+
+## Setting focus when the `popup` element is shown
+
+### The `delegatesfocus` attribute
+
+The `delegatesfocus` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that when the `popup` element is shown, focus should move to the first
+focusable descendent of the `popup` element.
+
+**Example:**
+
+```html
+<popup delegatesfocus>
+  <p>I am not a focusable element.</p>
+  <p>Nor am I.</p>
+  <button>I will be focused whenever the popup becomes focused.</button>
+</popup>
+```
+
+### Focusing steps
+
+The popup [focusing steps](https://html.spec.whatwg.org/multipage/interaction.html#focusing-steps)
+for a `popup` element _subject_ are as follows:
+
+1. If the `delegatesfocus` attribute is specified on _subject_, let _control_ be the first focusable
+   descendant element of _subject_, in [tree order](https://dom.spec.whatwg.org/#concept-tree-order),
+   that is not inert and has the
+   [autofocus](https://html.spec.whatwg.org/multipage/interaction.html#attr-fe-autofocus) attribute
+   specified.
+2. If there is no such descendent, but `delegatesfocus` is specified, let _control_ be _subject_.
+3. Otherwise, if the `autofocus` attribute is specified on _subject_, let _control_ be _subject_.
+4. Otherwise, return.
```

Comment:
returning from the algorithm.

Here, I'm not sure about WHATWG conventions. This is an if/elseif/else block, as step 4 is the "otherwise" for the embedded "if" in step 3. That, I haven't exactly seen.

Reply:
Any changes requested, e.g. melding 4 into step 3?